core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1962,6 +1962,20 @@ private[spark] object Utils extends Logging {
     path
   }
 
+  /**
+   * Updates Spark config with properties from a set of Properties.
+   * Provided properties have the highest priority.
+   */
+  def updateSparkConfigFromProperties(conf: SparkConf, properties: Map[String, String])
+    : Unit = {
+    properties.filter { case (k, v) =>
+      k.startsWith("spark.")
```

Comment:
You can move this to the `foreach` body.

On a side note, this whole method feels a little funny, but I have to see how it's used.